daemon/src/command.rs

```diff
@@ -35,6 +35,20 @@ impl Executor {
         }
     }
 
+    /// A way of querying values of a CFD, without issuing any commands or
+    /// mutating its state.
+    ///
+    /// Can be useful when preparing a command.
+    pub async fn query<T>(&self, id: OrderId, query: impl Fn(&Cfd) -> Result<T>) -> Result<T> {
+        let cfd = self
+            .db
+            .load_open_cfd(id, ())
+            .await
+            .context("Failed to load CFD")?;
+
+        query(&cfd).context("Failed to execute command on CFD")
+    }
```

Comment:
🤔 Doesn't seem like it would belong in the `Executor`.

My knowledge of event sourcing is limited, but I do think we have messed something up in the model when we have to query the aggregate for data that doesn't change throughout its lifetime. I hope this is improved if/when we split the order from the CFD.

In any case, I'm not blocking the PR on this. But I do think it sends the wrong message to consumers of `Executor` and I would prefer to do something ad-hoc like what I described in another comment.

Reply:
A couple of points why I don't see it as a hack:
- one needs to know *which* command to issue, or with what parameter to use it
- I tried to propose an alternative to the previous hack, with which we have gone to great lengths (piggybacking some data as part of result tuple that often doesn't change throughout its lifetime either, e.g. https://github.com/itchysats/itchysats/blob/master/model/src/cfd.rs#L780 )
The way I see it is that `query` or `peek` API to have a glance at values of CFD 
The latter ("peek") is common in functional programming, so one can decide whether they should consume/remap an iterator or not.

I totally agree that avoiding such change when we're under time pressure is a good idea, I'll try to pass the HashMap in! Might retry this when we run into this problem again at a more convenient time :-)

Reply:
> one needs to know which command to issue, or with what parameter to use it

I agree that we need to read things from the CFD. I just think it doesn't belong in the "command executor" module. Clearly this wasn't designed for reading from the CFD. 

Perhaps we could simply rebrand it or create a different wrapper around the db connection that let's us query the CFD. I'm not sure if it has any larger implications, because we are basically already doing this with `ExtractEventFromTuple`.

Reply:
fair enough, I see how it can be perceived as a hack!

I fully support an idea of some "query" wrapper for the DB; in most cases we actually needlessly pass in the whole DB, which can be written. I will look into this as a follow-up today to address the valuable feedback given here.